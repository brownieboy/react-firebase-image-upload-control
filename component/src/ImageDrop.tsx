import React, {useMemo} from "react";
import {useDropzone, DropzoneOptions} from "react-dropzone";

const baseStyle = {
  flex: 1,
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
  padding: "20px",
  borderWidth: 2,
  borderRadius: 2,
  borderColor: "#eeeeee",
  borderStyle: "dashed",
  backgroundColor: "#fafafa",
  color: "#bdbdbd",
  outline: "none",
  transition: "border .24s ease-in-out"
};

const activeStyle = {
  borderColor: "#2196f3"
};

const acceptStyle = {
  borderColor: "#00e676"
};

const rejectStyle = {
  borderColor: "#ff1744"
};

const StyledDropzone = (props: DropzoneOptions | undefined) => {
  const {
    getRootProps,
    getInputProps,
    isDragActive,
    isDragAccept,
    isDragReject
  } = useDropzone({
    accept: {"image/jpeg": [], "image/png": [], "image/gif": []},
    ...props
  });

  const style = useMemo(
    () => ({
      ...baseStyle,
      ...(isDragActive ? activeStyle : {}),
      ...(isDragAccept ? acceptStyle : {}),
      ...(isDragReject ? rejectStyle : {})
    }),
    [isDragActive, isDragAccept, isDragReject]
  );
  const message = props?.multiple
    ? "Drag 'n' drop some files here, or click to select files"
    : "Drag 'n' drop a file here, or click to select a file";
  return (
    <div className="container">
      {/* @ts-ignore */}
      <div {...getRootProps({style})}>
        <input {...getInputProps()} />
        <p>{message}</p>
      </div>
    </div>
  );
};

export default StyledDropzone;
