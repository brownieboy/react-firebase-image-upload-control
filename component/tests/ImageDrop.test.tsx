import React from "react";
import renderer from "react-test-renderer";
import {render} from "@testing-library/react";
import ImageDrop from "../src/ImageDrop";

describe("<ImageDrop />", () => {
  it("should show a component", () => {
    const component = <ImageDrop />;
    const {container} = render(component);
    expect(container.children.length).toBe(1);
  });

  it("should match the snapshot correctly", () => {
    const component = <ImageDrop />;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
