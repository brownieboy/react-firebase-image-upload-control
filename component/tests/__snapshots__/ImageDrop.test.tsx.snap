// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageDrop /> should match the snapshot correctly 1`] = `
<div
  className="container"
>
  <div
    onBlur={[Function]}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fafafa",
        "borderColor": "#eeeeee",
        "borderRadius": 2,
        "borderStyle": "dashed",
        "borderWidth": 2,
        "color": "#bdbdbd",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "outline": "none",
        "padding": "20px",
        "transition": "border .24s ease-in-out",
      }
    }
    tabIndex={0}
  >
    <input
      accept="image/*"
      multiple={true}
      onChange={[Function]}
      onClick={[Function]}
      style={
        {
          "display": "none",
        }
      }
      tabIndex={-1}
      type="file"
    />
    <p>
      Drag 'n' drop a file here, or click to select a file
    </p>
  </div>
</div>
`;
